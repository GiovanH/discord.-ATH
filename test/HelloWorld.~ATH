PROCREATE LOOP 1;
~ATH(LOOP) {
	input NAME "What's your name? ";
	print ("Hello, ~s\n!", NAME);
	LOOP.DIE();
}
THIS.DIE();