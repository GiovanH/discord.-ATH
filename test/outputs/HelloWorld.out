ProcreateStmt(VarExpr('LOOP'), IntExpr(1))
InputStmt(VarExpr('NAME'), StringExpr("What's your name? "))
StringExpr('Hello, ~s\\n!')
VarExpr('NAME')
PrintFunc([StringExpr('Hello, ~s\\n!'), VarExpr('NAME')])
KillFunc(VarExpr('LOOP'), [])
TildeAthLoop(VarExpr('LOOP'), Serialize([InputStmt(VarExpr('NAME'), StringExpr("What's your name? ")), PrintFunc([StringExpr('Hello, ~s\\n!'), VarExpr('NAME')]), KillFunc(VarExpr('LOOP'), [])], LOOP))
KillFunc(VarExpr('THIS'), [])
Graft(Serialize([ProcreateStmt(VarExpr('LOOP'), IntExpr(1)), TildeAthLoop(VarExpr('LOOP'), Serialize([InputStmt(VarExpr('NAME'), StringExpr("What's your name? ")), PrintFunc([StringExpr('Hello, ~s\\n!'), VarExpr('NAME')]), KillFunc(VarExpr('LOOP'), [])], LOOP)), KillFunc(VarExpr('THIS'), [])], THIS), 33)
What's your name? Hello, theSardonyx
!{'THIS': BuiltinSymbol(), 'NULL': BuiltinSymbol(), 'DIE': BuiltinSymbol(), 'ATH': BuiltinSymbol(), 'print': BuiltinSymbol(), 'input': BuiltinSymbol(), 'import': BuiltinSymbol(), 'DEBATE': BuiltinSymbol(), 'UNLESS': BuiltinSymbol(), 'EXECUTE': BuiltinSymbol(), 'DIVULGATE': BuiltinSymbol(), 'FABRICATE': BuiltinSymbol(), 'REPLICATE': BuiltinSymbol(), 'PROCREATE': BuiltinSymbol(), 'BIFURCATE': BuiltinSymbol(), 'AGGREGATE': BuiltinSymbol(), 'LOOP': AthSymbol(False, 1, None)}
